products-service/src/functions/getProductById/handler.test.ts

```diff
@@ -0,0 +1,65 @@
+import { main as handler } from './handler';
+
+import mockEvent from '@serverless/event-mocks';
+
+import { getProductById } from '@services/products';
+import { APIGatewayProxyEvent, Context } from 'aws-lambda';
+import { STATUS_500_ERROR_MESSAGE } from '@libs/lambda';
+
+jest.mock('@services/products');
+
+const getProductByIdMock = getProductById as unknown as jest.Mock<() => any>;
+
+describe('getProductsById', () => {
+    afterEach(jest.clearAllMocks);
+
+    const getEventStub = (productId) => mockEvent(
```

Comment:
I'd not cover infrastructure logic with tests. 
Instead, I'd focus on business logic, because it can run on any infrastructure.

Reply:
I don't think that this is "infrastructure logic" (please correct me if I wrong). "Infrastructure tests" mean that we are running our code on particular server under particular OS etc. Unit tests cannot be infrastructure by default. These particular tests are only for controller, with their help I can ensure that user will get only correct and safe responses. And will not send unsafe, such as stack traces with confidential data.

I strongly agree that service should work correctly and return only correct data as well (all exceptions should be handled, etc). But we have documentation, and we want to ensure, that our endpoint will send correct status codes. I believe in such cases it's good point to write unit tests, especially for long-term projects. With these unit tests we don't need to create infrastructure tests.

And service and business logic should not know about HTTP statuses etc.

I didn't cover services itself with tests because they are mocks. "There is no spoon".